diff --git a/VTK/IO/Geometry/vtkSTLReader.cxx b/VTK/IO/Geometry/vtkSTLReader.cxx
index e6b0f18..4016e57 100644
--- a/VTK/IO/Geometry/vtkSTLReader.cxx
+++ b/VTK/IO/Geometry/vtkSTLReader.cxx
@@ -448,7 +448,7 @@ bool vtkSTLReader::ReadASCIISTL(FILE *fp, vtkPoints *newPts,
           done = done || (fscanf(fp,"%s", line)==EOF);
           }
         }
-      if (!done)
+      else if (!done)
         {
         done = (fgets(line, 255, fp) == 0);
         lineCount++;
