diff --git a/ParaViewCore/ServerImplementation/Core/CMakeLists.txt b/ParaViewCore/ServerImplementation/Core/CMakeLists.txt
index 36f5f97..5dd0111 100644
--- a/ParaViewCore/ServerImplementation/Core/CMakeLists.txt
+++ b/ParaViewCore/ServerImplementation/Core/CMakeLists.txt
@@ -94,7 +94,7 @@ endif()
 add_custom_command(
     OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/vtkPVMessage.pb.h
            ${CMAKE_CURRENT_BINARY_DIR}/vtkPVMessage.pb.cc
-    
+
     COMMAND ${_PROTOC_COMPILER}
            "--cpp_out=dllexport_decl=VTKPVSERVERIMPLEMENTATIONCORE_EXPORT:${CMAKE_CURRENT_BINARY_DIR}"
            --proto_path "${CMAKE_CURRENT_SOURCE_DIR}"
@@ -105,7 +105,7 @@ add_custom_command(
 
 add_custom_target(
   protobuf_code_generation
-  DEPENDS 
+  DEPENDS
     ${CMAKE_CURRENT_BINARY_DIR}/vtkPVMessage.pb.h
     ${CMAKE_CURRENT_BINARY_DIR}/vtkPVMessage.pb.cc
 )
@@ -118,5 +118,16 @@ endif()
 vtk_module_library(vtkPVServerImplementationCore
   ${Module_SRCS})
 
+if (VTK_USE_SYSTEM_PROTOBUF)
+  # Protobuf 3.1 doesn't decorate some of its symbols with the _EXPORT macro,
+  # so they are not visible when using `-fvisibility=hidden`. Explicitly set
+  # the visibility to `default` to work around this. It is fixed as of
+  # protobuf-3.2.0-rc1 by this commit:
+  # <https://github.com/google/protobuf/commit/b1295eeffbf2ae7da8a5c7d8601c3ace0663fa09>.
+  set_target_properties(vtkPVServerImplementationCore
+    PROPERTIES
+      CXX_VISIBILITY_PRESET default)
+endif ()
+
 add_dependencies(vtkPVServerImplementationCore
   protobuf_code_generation)
diff --git a/ParaViewCore/ServerImplementation/Core/vtkPVMessage.proto b/ParaViewCore/ServerImplementation/Core/vtkPVMessage.proto
index aed5832..3868506 100644
--- a/ParaViewCore/ServerImplementation/Core/vtkPVMessage.proto
+++ b/ParaViewCore/ServerImplementation/Core/vtkPVMessage.proto
@@ -1,6 +1,8 @@
 // ****************************************************************************
 //                    vtkInternalDataStreamIO.proto
 // ****************************************************************************
+syntax = "proto2";
+
 package paraview_protobuf;
 
 // Type of generated code : [SPEED, CODE_SIZE, LITE_RUNTIME]
