--- a/applications/utilities/postProcessing/graphics/PVReaders/PVblockMeshReader/PVblockMeshReader/CMakeLists.txt
+++ b/applications/utilities/postProcessing/graphics/PVReaders/PVblockMeshReader/PVblockMeshReader/CMakeLists.txt
@@ -10,8 +10,11 @@
 CMAKE_MINIMUM_REQUIRED(VERSION 2.6)
 
 FIND_PACKAGE(ParaView REQUIRED)
+FIND_PACKAGE(Qt4 REQUIRED)
 INCLUDE(${PARAVIEW_USE_FILE})
 
+
+
 LINK_DIRECTORIES(
     $ENV{FOAM_LIBBIN}
     $ENV{FOAM_EXT_LIBBIN}
@@ -42,13 +45,7 @@ SET(
 #
 
 # Extend the auto-generated panel
-QT4_WRAP_CPP(MOC_SRCS pqPVblockMeshReaderPanel.h)
-
-ADD_PARAVIEW_OBJECT_PANEL(IFACES IFACE_SRCS
-  CLASS_NAME pqPVblockMeshReaderPanel
-  XML_NAME  PVblockMeshReader # name of SourceProxy in *SM.xml
-  XML_GROUP sources
-)
+QT4_WRAP_CPP( MOC_SRCS pqPVblockMeshReaderPanel.h )
 
 IF("${PARAVIEW_VERSION_MAJOR}.${PARAVIEW_VERSION_MINOR}" EQUAL 4.0)
 
